### PR TITLE
Add support for symlink_outputs to Blueprint

### PR DESCRIPTION
**Pull request by jin**

### CODE REVIEW

**Comment by jin** [comment]
cc @colincross @danw @lberki

File: ninja_defs.go [hunk 1]
```diff
@@ -56,20 +56,22 @@ type PoolParams struct {
...
+	Restat         bool     // Whether Ninja should re-stat the rule's outputs.
+	Rspfile        string   // The response file.
+	RspfileContent string   // The response file content.
+	SymlinkOutputs []string // The list of symlink output targets.
 
 	// These fields are used internally in Blueprint
 	CommandDeps      []string // Command-specific implicit dependencies to prepend to builds
 	CommandOrderOnly []string // Command-specific order-only dependencies to prepend to builds
 	Comment          string   // The comment that will appear above the definition.
+
```
Comment (colincross) [inline ninja_defs.go:74]:
```suggestion
```

File: ninja_defs.go [hunk 2]
```diff
@@ -56,20 +56,22 @@ type PoolParams struct {
...
+	Command        string   // The command that Ninja will run for the rule.
+	Depfile        string   // The dependency file name.
+	Deps           Deps     // The format of the dependency file.
+	Description    string   // The description that Ninja will print for the rule.
+	Generator      bool     // Whether the rule generates the Ninja manifest file.
+	Pool           Pool     // The Ninja pool to which the rule belongs.
+	Restat         bool     // Whether Ninja should re-stat the rule's outputs.
+	Rspfile        string   // The response file.
+	RspfileContent string   // The response file content.
+	SymlinkOutputs []string // The list of symlink output targets.
```
Comment (colincross) [inline ninja_defs.go:68]:
```suggestion
	SymlinkOutputs []string // The list of Outputs or ImplicitOutputs that are symlinks.
```

File: ninja_defs.go [hunk 3]
```diff
@@ -84,6 +86,7 @@ type BuildParams struct {
 	Rule            Rule              // The rule to invoke.
 	Outputs         []string          // The list of explicit output targets.
 	ImplicitOutputs []string          // The list of implicit output targets.
+	SymlinkOutputs  []string          // The list of symlink output targets.
```
Comment (colincross) [inline ninja_defs.go:89]:
```suggestion
	SymlinkOutputs []string // The list of Outputs or ImplicitOutputs that are symlinks.
```

File: ninja_defs.go [hunk 4]
```diff
@@ -205,6 +207,15 @@ func parseRuleParams(scope scope, params *RuleParams) (*ruleDef,
 		r.Variables["rspfile_content"] = value
 	}
 
+	if len(params.SymlinkOutputs) > 0 {
+		value, err = parseNinjaString(scope, strings.Join(params.SymlinkOutputs, " "))
+		if err != nil {
+			return nil, fmt.Errorf("error parsing symlink_outputs param: %s",
```
Comment (lizkammer) [inline ninja_defs.go:213]: Optional:

```suggestion
			return nil, fmt.Errorf("error parsing SymlinkOutputs param: %s",
```

**Review by lizkammer — APPROVED** [review]

File: ninja_defs.go (hunk 4 above)
Comment (jin) [reply ninja_defs.go]: Done.

**Pull request merged**